viadot/sources/redshift_spectrum.py

```diff
@@ -0,0 +1,199 @@
+
+    def from_df(
+        self,
+        df: pd.DataFrame,
+        to_path: str,
+        database: str,
+        table: str,
+        extension: str = ".parquet",
+        if_exists: Literal["overwrite", "append"] = "overwrite",
+        partition_cols: List[str] = None,
+        index: bool = False,
+        compression: str = None,
+        sep: str = ",",
+        description: str = "test",
+    ) -> None:
+        """
+        Upload a pandas `DataFrame` to a csv or parquet file.
+
+        Args:
+            df (pd.DataFrame): Pandas DataFrame.
+            to_path (str): Path to a S3 folder where the table will be located.
+                Defaults to None.
+            extension (str): Required file type. Accepted file formats are 'csv'
+                and 'parquet'.
+            database (str): AWS Glue catalog database name.
+            table (str): AWS Glue catalog table name.
+            partition_cols (List[str]): List of column names that will be used to
+                create partitions. Only takes effect if dataset=True.
+            if_exists (str, optional): 'overwrite' to recreate any possible existing
+                table or 'append' to keep any possible existing table. Defaults to
+                overwrite.
+            index (bool, optional): Write row names (index). Defaults to False.
+            compression (str, optional): Compression style (None, snappy, gzip, zstd).
+            sep (str, optional): Field delimiter for the output file. Defaults to ','.
+            description (str, optional): AWS Glue catalog table description.
+        """
+
+        if extension == ".parquet":
+            wr.s3.to_parquet(
+                boto3_session=self.session,
+                df=df,
+                path=to_path,
+                mode=if_exists,
+                index=index,
+                compression=compression,
+                dataset=True,
+                database=database,
+                table=table,
+                partition_cols=partition_cols,
+                description=description,
+            )
+        elif extension == ".csv":
+            wr.s3.to_csv(
+                boto3_session=self.session,
+                df=df,
+                path=to_path,
+                dataset=True,
+                database=database,
+                table=table,
+                index=index,
+                sep=sep,
+            )
+        else:
+            raise ValueError("Only CSV and parquet formats are supported.")
+
+    def to_df(
+        self,
+        database: str,
+        table: str,
+    ) -> pd.DataFrame:
+        """
+        Reads a Spectrum table to a pandas `DataFrame`.
+
+        Args:
+            database (str): AWS Glue catalog database name.
+            table (str): AWS Glue catalog table name.
+        """
+
+        df = wr.s3.read_parquet_table(
+            boto3_session=self.session,
+            database=database,
+            table=table,
+        )
```

Comment:
It seems like this reads Glue, not Redshift. In theory, `database` can be different than the name of the schema for this table in Redshift.